mousestyles/classification.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import print_function, absolute_import, division
+
+import os
+import time
+import numpy as np
+
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.linear_model import LogisticRegression
+from sklearn.naive_bayes import GaussianNB
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.ensemble import ExtraTreesClassifier
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.svm import LinearSVC
+from sklearn.svm import SVC
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
+from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
+from sklearn.cross_validation import cross_val_predict
+from sklearn.cross_validation import StratifiedKFold
+from sklearn.metrics import accuracy_score
+
+from mousestyles import data_dir
+
+data_path = os.path.join(data_dir, "all_features_mousedays_11bins.npy")
+seed = 1
+model_list = [KNeighborsClassifier,
+              LogisticRegression,
+              GaussianNB,
+              RandomForestClassifier,
+              ExtraTreesClassifier,
+              GradientBoostingClassifier,
+              LinearSVC,
+              SVC,
+              LinearDiscriminantAnalysis,
+              QuadraticDiscriminantAnalysis]
+
+models_params = {
+    "KNeighborsClassifier": {
+        "n_neighbors": 5,
+        "weights": 'uniform',
+        "n_jobs": 1,
+    },
+    "LogisticRegression": {
+        "solver": "lbfgs",
+        "multi_class": "multinomial",
+        "penalty": "l2",
+        "C": 1.0,
+    },
+    "GaussianNB": {
+    },
+    "RandomForestClassifier": {
+        "n_estimators": 500,
+        "criterion": "gini",
+        "max_depth": 8,
+        "bootstrap": True,
+        "random_state": seed,
+        "verbose": 0,
+        "n_jobs": -1,
+    },
+    "ExtraTreesClassifier": {
+        "n_estimators": 500,
+        "criterion": "gini",
+        "max_depth": 8,
+        "bootstrap": True,
+        "random_state": seed,
+        "verbose": 0,
+        "n_jobs": -1,
+    },
+    "GradientBoostingClassifier": {
+        "loss": "deviance",
+        "learning_rate": 0.1,
+        "n_estimators": 100,
+        "subsample": 1.0,
+        "max_depth": 6,
+        "random_state": seed,
+        "max_features": 10,
+        "verbose": 0,
+    },
+    "LinearSVC": {
+        "penalty": "l2",
+        "loss": "hinge",
+        "C": 1.0,
+        "verbose": 0,
+        "random_state": seed,
+        "multi_class": "ovr",
+    },
+    "SVC": {
+        "C": 1.0,
+        "kernel": "rbf",
+        "gamma": .01,
+        "random_state": seed,
+        "verbose": 0,
+    },
+    "LinearDiscriminantAnalysis": {
+        "solver": "lsqr",
+        "shrinkage": "auto",
+    },
+    "QuadraticDiscriminantAnalysis": {
+        "reg_param": .1,
+    },
+}
+
+
+def load_data(data_path):
```

Comment:
@hduongtrong this function should probably be moved to the data module (or, it may be a duplicate of something that's already there).